### DISCUSSION TITLE
Question: Export database to a buffer

### DISCUSSION THREAD

**Discussion by Poliziano** [vlcn-io/cr-sqlite#266] [Q&A]
I have been using [sql.js](https://sql.js.org/#/) for a web app that saves the SQLite file to the user's disk. I would like to switch over to vulcan and would like to know if there is a simple way to export from [@vlcn.io/crsqlite-wasm](https://www.npmjs.com/package/@vlcn.io/crsqlite-wasm) similar to sql.js' [export](https://sql.js.org/#/?id=write-a-database-to-the-disk)?

**Comment by tantaman** [comment]
I haven't implemented import/export yet although some people were interested in doing it not too long ago and the details of how to do it were discussed here: https://github.com/rhashimoto/wa-sqlite/discussions/82

There's also a chance we'll support the [official SQLite WASM port](https://sqlite.org/wasm/doc/trunk/index.md) in the future. The reason that is relevant here is that the official build includes db import/export support.

**Comment by tantaman** [comment]
There's this demo of db import and export here:
https://rhashimoto.github.io/wa-sqlite/demo/file/

source code: https://github.com/rhashimoto/wa-sqlite/tree/master/demo/file
